packages/core/src/App/Containers/VerificationModal/__test__/verification-modal.spec.tsx

```diff
@@ -0,0 +1,90 @@
+    isMobile: jest.fn(),
+    isDesktop: jest.fn(),
+}));
+
+jest.mock('../verification-modal-content', () => jest.fn(() => <div>VerificationModalContent</div>));
+
+describe('<VerificationDocumentSubmited />', () => {
+    let modal_root_el: HTMLDivElement;
+    const mock_store = mockStore({});
+
+    beforeAll(() => {
+        modal_root_el = document.createElement('div');
+        modal_root_el.setAttribute('id', 'modal_root');
+        document.body.appendChild(modal_root_el);
+    });
+
+    afterAll(() => {
+        document.body.removeChild(modal_root_el);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should not render the VerificationModal component', async () => {
+        mock_store.ui.is_verification_modal_visible = true;
+        const title = 'Submit your proof of identity and address';
+        render(
+            <StoreProvider store={mock_store}>
+                <VerificationModal />
+            </StoreProvider>
+        );
+        expect(screen.queryByText(title)).not.toBeInTheDocument();
+        expect(screen.queryByText('VerificationModalContent')).not.toBeInTheDocument();
+    });
+
+    it('should render the VerificationModal component', async () => {
+        (isDesktop as jest.Mock).mockReturnValue(true);
+        mock_store.ui.is_verification_modal_visible = true;
+        const title = 'Submit your proof of identity and address';
+        render(
+            <StoreProvider store={mock_store}>
+                <VerificationModal />
+            </StoreProvider>
+        );
+        expect(screen.getByText(title)).toBeInTheDocument();
+        expect(screen.getByText('VerificationModalContent')).toBeInTheDocument();
+    });
+
+    it('should setIsVerificationModalVisible to false', async () => {
+        (isDesktop as jest.Mock).mockReturnValue(true);
+        mock_store.ui.is_verification_modal_visible = true;
+        render(
+            <StoreProvider store={mock_store}>
+                <VerificationModal />
+            </StoreProvider>
+        );
+        const close_button = screen.getByRole('button');
+        expect(close_button).toBeInTheDocument();
+        fireEvent.click(close_button);
+        expect(mock_store.ui.setIsVerificationModalVisible).toHaveBeenCalledWith(false);
+    });
+
+    it('should setIsVerificationModalVisible to be false in isMobile', async () => {
+        modal_root_el = document.createElement('div');
+        modal_root_el.setAttribute('id', 'deriv_app');
+        document.body.appendChild(modal_root_el);
+
+        (isDesktop as jest.Mock).mockReturnValue(false);
+        (isMobile as jest.Mock).mockReturnValue(true);
+        mock_store.ui.is_verification_modal_visible = true;
+        render(
+            <StoreProvider store={mock_store}>
+                <VerificationModal />
+            </StoreProvider>
+        );
+        const close_button = screen.getByTestId('dt-dc-mobile-dialog-close-btn');
```

Comment:
getByRole?

Reply:
Tried getByRole, seems like the component uses `div` instead of `button` so can't target the element using getByRole